src/lib/entities/user/model/store.ts

```diff
@@ -0,0 +1,5 @@
+import { writable } from 'svelte/store';
+import type { User } from '@supabase/supabase-js';
+import { supabaseClient } from '$lib/shared/api';
+
+export const user = writable<User | null>(supabaseClient.auth.user());
```

Comment:
Question: in SvelteKit, does this line mean that authentication will happen on the server or after the page is loaded on client? Or, in other words, will it cause the FOUC?

Reply:
It's not gonna cause FOUC because it has nothing to do with CSS, but it will render as logged out first and then catch up immediately. I think we can do better, but that's probably going to require some research about how CloudFlare SSR works. I think it's worth another issue

Reply:
Ok, I understand now. I guess the term [Cumulative Layout Shift (CLS)](https://web.dev/cls/) describes the situation better